File: pegasusX/apps/supplier-app-ios/SupplierApp/Views/Planning/SeasonalOverridesList.swift
import SwiftUI

struct SeasonalOverridesList: View {
    let overrides: [SeasonalOverrideRow]

    var body: some View {
        if overrides.isEmpty {
            Text("mobile_supplier.ui.no_custom_seasonal_overrides_yet")
                .foregroundStyle(.secondary)
        } else {
            ForEach(overrides) { row in
                VStack(alignment: .leading, spacing: SupplierTheme.spacingXS) {
                    Text(row.name?.isEmpty == false ? row.name! : row.templateId)
                        .font(.headline)
                    Text(L10n.format("mobile_supplier.ui.startdate_enddate", "\(row.startDate)", "\(row.endDate)"))
                        .font(.caption)
                        .foregroundStyle(.secondary)
                    if let m = row.multiplier {
                        Text("×\(m)")
                            .font(.caption2)
                            .foregroundStyle(.secondary)
                    }
                    Text(row.isActive ? "Active" : "Inactive")
                        .font(.caption2)
                        .foregroundStyle(row.isActive ? SupplierTheme.success : SupplierTheme.secondaryLabel)
                }
            }
        }
    }
}

File: pegasusX/apps/supplier-app-android/app/src/main/java/com/pegasusx/supplier/ui/screens/planning/SeasonalOverridesList.kt
package com.pegasusx.supplier.ui.screens.planning

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.pegasusx.supplier.data.model.SeasonalOverrideRow
import com.pegasusx.supplier.ui.components.SupplierOpsListCard
import com.pegasusx.supplier.ui.theme.PegasusSpacing

@Composable
fun SeasonalOverridesList(overrides: List<SeasonalOverrideRow>) {
    Column(verticalArrangement = Arrangement.spacedBy(PegasusSpacing.md), modifier = Modifier.fillMaxWidth()) {
        if (overrides.isEmpty()) {
            Text(
                "No custom seasonal overrides yet.",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        } else {
            overrides.forEach { row ->
                SupplierOpsListCard(
                    headline = row.name?.ifBlank { row.templateId } ?: row.templateId,
                    supporting = "${row.startDate} → ${row.endDate} · ×${row.multiplier} · ${if (row.isActive) "Active" else "Inactive"}",
                )
            }
        }
    }
}
